src/utils/proxyMap.ts

```diff
@@ -0,0 +1,113 @@
+import { proxy } from 'valtio'
+
+type KeyValRecord<K, V> = {
+  key: K
+  value: V
+}
+
+type InternalProxyMap<K, V> = Map<K, V> & {
+  data: KeyValRecord<K, V>[]
+  toJSON: object
+}
+
+/**
+ * proxyMap
+ *
+ * This is to create a proxy which mimic the native Map behavior.
+ * The API is the same as Map API
+ *
+ * @example
+ * import { proxyMap } from 'valtio/utils'
+ * const state = proxyMap([["key", "value"]])
+ *
+ * //can be used inside a proxy as well
+ * const state = proxy({
+ *   count: 1,
+ *   map: proxyMap()
+ * })
+ */
+export const proxyMap = <K, V>(
+  entries: Iterable<readonly [K, V]> | null = []
```

Comment:
can be this?
```suggestion
  entries?: Iterable<readonly [K, V]> | null = []
```

Reply:
Given it has a default value `[]` we don't have to put the question mark
![Screenshot 2022-01-20 at 08 29 40](https://user-images.githubusercontent.com/5574267/150292998-e2454e1c-f487-4bc5-a0d5-ecdfbae2ca5d.png)

Reply:
My bad!